### PR TITLE
Version 0.3-BETA

### PR DESCRIPTION
**Pull request by TureBentzin**
Important additions:

1. New "Motion" Features
2. Log4j Feature
3. Inventory Feature
4. CompletableMessages
5. new Utils
6. UI and Message polishing
7. fixed some bugs

### CODE REVIEW

**Comment by TureBentzin** [comment]
This version brings primarily these new features:
- MotionFeature
- InventoryFeature
- Log4JFeature
- "replaceable message arguments" -> CompletableMessage

and a lot of small fixes and polishing needed for further development

File: src/main/java/de/bentzin/ingwer/command/IngwerCommand.java [hunk 1]
```diff
@@ -84,8 +86,7 @@ public Collection<CommandTarget> commandTargetCollection() {
     @ApiStatus.Experimental
     public <T extends IngwerCommand> Identity identityCommand(IngwerCommandSender commandSender, @NotNull CommandTarget senderType, Consumer<Identity> action) {
         if (senderType.equals(CommandTarget.INGAME)) {
-            if (commandSender instanceof Identity) {
-                Identity identity = (Identity) commandSender;
+            if (commandSender instanceof Identity identity) {
```
Comment (Bommels05) [inline src/main/java/de/bentzin/ingwer/command/IngwerCommand.java:89]: : ComNoKotlin :

File: src/main/java/de/bentzin/ingwer/command/internalcommands/ChatCommand.java [hunk 2]
```diff
@@ -28,8 +28,10 @@ public void execute(IngwerCommandSender commandSender, String @NotNull [] cmd, C
             OneLinedMessage build = chatMessageBuilder(commandSender).add(gen(cmd)).build();
             Objects.requireNonNull(Ingwer.getStorage().getAllIdentities()).forEach(build::send);
 
-        } else {
+        } else //noinspection SpellCheckingInspection
+        {
             //TODO: bommels05
+            //noinspection SpellCheckingInspection
             if (commandSender.getName().equals("Bommels05")) {
```
Comment (Bommels05) [inline src/main/java/de/bentzin/ingwer/command/internalcommands/ChatCommand.java:35]: I am not deaf

File: src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+    }
+
+    @Override
+    public void onDisable() {
+        glitch_players.clear();
+        drunk_players.clear();
+        silkworm.stop();
+        bukkitTask.cancel();
+        getLogger().warning("WARN: DEBUG FEATURE SHUTDOWN!!!");
```
Comment (Bommels05) [inline src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java:82]: So this is a Debug Feature?

File: src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        return new Silkworm() {
+            @Override
+            public void run() {
+                for (Player player : Bukkit.getOnlinePlayers()) {
+                    if (drunk_players.contains(player.getUniqueId()))
+                        if (player != null) {
+                            float yaw = player.getLocation().getYaw();
+                            Vector vector = generateMotion(player.getUniqueId(), yaw);
+                            applyMotion(player, vector);
+                            getLogger().info(String.valueOf(yaw));
```
Comment (Bommels05) [inline src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java:201]: I think this is a debug

File: src/main/java/de/bentzin/ingwer/features/integrated/Log4JFeature.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            });
+        }
+
+        @Override
+        public CommandTarget[] getCommandTargets() {
+            return new CommandTarget[]{CommandTarget.INGAME};
+        }
+
+        /**
+         * Benny
```
Comment (Bommels05) [inline src/main/java/de/bentzin/ingwer/features/integrated/Log4JFeature.java:109]: Benny?

**Review by Bommels05 — APPROVED** [review]
This PR polishes a lot of Things and is a good and right step in the development of ingwer!

File: src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java (hunk 3 above)
Comment (TureBentzin) [reply src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java:82]:
```suggestion
```
yes

File: src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java (hunk 4 above)
Comment (TureBentzin) [reply src/main/java/de/bentzin/ingwer/features/integrated/DrunkMotionFeature.java:200]:
```suggestion
```
yes

File: src/main/java/de/bentzin/ingwer/features/integrated/Log4JFeature.java (hunk 5 above)
Comment (TureBentzin) [reply src/main/java/de/bentzin/ingwer/features/integrated/Log4JFeature.java:109]: simply because benny was there and it started working...

**Pull request merged**